### DISCUSSION TITLE
Integration with Jboss EAP 7

### DISCUSSION THREAD

**Discussion by irizzant** [vladimir-bukhtoyarov/bucket4j#264] [Q&A] [answered]
Is there any example that could guide me to integrate bucket4j with Jboss EAP 7 using Infinispan?

**Comment by vladimir-bukhtoyarov** [comment]
Hello @irizzant

There are no such examples. Bucket4j is just a library that does not require any special from your application environment, so Jboss usage scenario does not differ from any other use-case. Try to start from this page https://bucket4j.com/7.5.0/toc.html#bucket4j-infinispan and feel free to ask more special questions in case of problems.

**Comment by irizzant** [comment, marked as answer]
Hello @vladimir-bukhtoyarov 
that's where I started from, but I don't see anything showing how to hook into the integrated Infinispan cache in Jboss and get the following: 
```
org.infinispan.functional.FunctionalMap.ReadWriteMap<K, byte[]> map = ...;
private static final InfinispanProxyManager<K> proxyManager = new InfinispanProxyManager(map);
```

**Reply by vladimir-bukhtoyarov** [reply]
If you have access to ```org.infinispan.Cache``` then functional map can be created from it via following code:
```java
private static ReadWriteMap<String, byte[]> toMap(org.infinispan.Cache<String, byte[]> cache) {
        FunctionalMapImpl<String, byte[]> functionalMap = FunctionalMapImpl.create(cache.getAdvancedCache());
        return ReadWriteMapImpl.create(functionalMap);
}
```
If [this code example](https://github.com/bucket4j/bucket4j/blob/master/bucket4j-infinispan-all/bucket4j-infinispan/src/test/java/io/github/bucket4j/grid/infinispan/InfinispanTest.java) is not helpful then it is better to ask JBoss/Infinispan authours directly on their forums.
